controller_manager/src/ros2_control_node.cpp

```diff
@@ -67,10 +67,22 @@ int main(int argc, char ** argv)
     RCLCPP_WARN(cm->get_logger(), "Unable to lock the memory : '%s'", message.c_str());
   }
 
-  const int cpu_affinity = cm->get_parameter_or<int>("cpu_affinity", -1);
-  if (cpu_affinity >= 0)
+  rclcpp::Parameter cpu_affinity_param;
+  if (cm->get_parameter("cpu_affinity", cpu_affinity_param))
   {
-    const auto affinity_result = realtime_tools::set_current_thread_affinity(cpu_affinity);
+    std::vector<int> cpus = {};
+    if (cpu_affinity_param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER)
+    {
+      cpus = {cpu_affinity_param.as_int()};
+    }
+    else if (cpu_affinity_param.get_type() == rclcpp::ParameterType::PARAMETER_INTEGER_ARRAY)
```

Comment:
You might want to update the documentation: https://github.com/ros-controls/ros2_control/blob/master/controller_manager/doc/userdoc.rst

Reply:
Thank you for reminding!